File: test_resources/kotlin/struct_generics.kt
data class GenericType(
    val items: List<Test>,
    val table: LinearTable<Float, Test>,
) {
    companion object {
        fun createDefault(): GenericType = GenericType(
            items = listOf(),
            table = LinearTable.createDefault<Float, Test>(),
        )

        fun readFromBuffers(reader: Long): GenericType {
            val items = readFromBuffersList(reader) {
                Test.readFromBuffers(reader)
            }
            val table = LinearTable<Float, Test>.readFromBuffers(reader)

            return GenericType(
                items = items,
                table = table,
            )
        }

        fun skipInBuffers(reader: Long, count: Int) {
            for (i in 0..<count) {
                readFromBuffersList(reader) {
                    Test.readFromBuffers(reader)
                }
                LinearTable<Float, Test>.readFromBuffers(reader)
            }
        }
    }

    fun writeToBuffers(writer: Long) {
        writeToBuffersList(writer, items) { itemsItem ->
            itemsItem.writeToBuffers(writer)
        }
        table.writeToBuffers(writer)
    }
}

File: test_resources/dart/struct_generics.dart
class GenericType {
  const GenericType({
    required this.items,
    required this.table,
  });

  const GenericType.createDefault()
      : items = const <Test>[],
        table = const LinearTable<double, Test>.createDefault();

  final List<Test> items;
  final LinearTable<double, Test> table;
}

class GenericTypeBuffersFactory implements BuffersFactory<GenericType> {
  const GenericTypeBuffersFactory();

  @override
  GenericType createDefault() => const GenericType.createDefault();
}
